kroxylicious-filters/kroxylicious-record-encryption/src/main/java/io/kroxylicious/filter/encryption/RecordEncryptionFilter.java

```diff
@@ -169,11 +173,28 @@ public CompletionStage<ResponseFilterResult> onFetchResponse(short apiVersion, R
         return maybeDecodeFetch(response.responses(), context)
                 .thenCompose(responses -> context.forwardResponse(header, response.setResponses(responses)))
                 .exceptionallyCompose(throwable -> {
-                    log.atWarn().setMessage("failed to decrypt records, cause message: {}")
-                            .addArgument(throwable.getMessage())
-                            .setCause(log.isDebugEnabled() ? throwable : null)
-                            .log();
-                    return CompletableFuture.failedStage(throwable);
+                    if (throwable.getCause() instanceof KmsException) {
```

Comment:
Would (both branches) of this be observable from metrics alone?

Reply:
I think the generation of a KMSException is going to pass through: 
https://github.com/kroxylicious/kroxylicious/blob/574f67316e69831f259df689f12865d922ab2ffb/kroxylicious-filters/kroxylicious-record-encryption/src/main/java/io/kroxylicious/filter/encryption/kms/InstrumentedKms.java#L46-L52

so ultimately yes it will be in metrics.

Reply:
I've been thinking more a out this. Given the use case and the severity of the problem I think we should have a dedicated metric. So that it can be alerted on directly.

I was also wondering if we could tag a counter with the topic name and offset of the problem, but that leads to counter explosion, so maybe an info gauge (like the build info). We can update it once with the whole offset range we encounter.